### PR TITLE
options/glibc: Add iopl, ioperm and the port i/o functions

### PR DESCRIPTION
**Pull request by Dennisbonke**
More Valgrind additions.

Part of the mlibc LFS project.

### CODE REVIEW

File: options/glibc/generic/sys-io.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+int ioperm(unsigned long int, unsigned long int, int) {
+	__ensure(!"Not implemented");
+	__builtin_unreachable();
+}
+
+int iopl(int) {
+	__ensure(!"Not implemented");
+	__builtin_unreachable();
+}
```
Comment (ArsenArsen) [inline options/glibc/generic/sys-io.cpp:12]: both of these are x86(_64) specific and syscall wrappers, stubbing them is extremely silly

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
is this header x86(_64) only? I'm unsure

File: options/glibc/include/sys/io.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int ioperm(unsigned long int from, unsigned long int num, int turn_on);
+
+int iopl(int level);
+
+#ifdef __x86_64__
+static __inline unsigned char inb(unsigned short int port) {
```
[the remark this answers is not in the document]
Comment (64) [reply options/glibc/include/sys/io.h:13]: (= `__MLIBC_INLINE_DEFINITION`)

**Comment by no92** [comment]
Rebased onto current master and addressed the comments raised about the header.

**Comment by no92** [comment]
Implemented `ioperm` and `iopl` with sysdeps for Linux, also rebased on master.

**Review by Geertiebear — APPROVED** [review]
LGTM

**Pull request merged**